### PR TITLE
More work on the Reference Manual

### PR DESCRIPTION
**Pull request by ffesti**
Mainly reformat and improve the Tags page.

### CODE REVIEW

**Comment by ffesti** [comment]
http://ffesti.github.io/rpm/manual/tags.html for the rendered result.

File: docs/manual/tags.md [hunk 1]
```diff
@@ -2,7 +2,38 @@
...
+### Scalar types
+
+There are four unsigned integer types for RPM tags:
+
+* `int8`
+* `int16`
+* `int32`
+* `int64`
+
+`int8` and `int16` are rarely used. Most integer tags are `int32` with a few `int64` replacements where 32 bits were just not enough. These being unsigned limits time stamps to 1970-01-01 and later.
```
Comment (pmatilai) [inline docs/manual/tags.md:20]: Leave this paragraph out, eg the timestamp mention is only confusing in this context, and whether a type is common or not isn't really relevant. I'd rather point out that these are technically always arrays.

File: docs/manual/tags.md [hunk 2]
```diff
@@ -285,6 +285,10 @@ Sigsize           | 257  | int32        | Header+payload size.
 
 ## Installed packages
 
+Some information on how exactly a package got installed can not be
+known before hand. RPM though adds a few tags to the header during
+installation before adding it to the rpmdb.
```
Comment (pmatilai) [inline docs/manual/tags.md:290]: Some grammar/spelling issues in this paragraph. I'd go with just "The following tags do not exist in package files, they are only ever added during installation." or something like that.

Comment (DemiMarie) [reply docs/manual/tags.md:290]: Does RPM correctly reject such tags in package files?

File: docs/manual/tags.md [hunk 3]
```diff
@@ -344,259 +182,189 @@ denotes the scriptlet interpreter and possible arguments, and flags tag
...
+Transfiletriggerversion     | 5081 | string array
 
-* Sigsize (257) int32
+## Signatures and digests
 
-  Header+payload size.
+[Signatures](signatures.md) allow to verify the origin of a package.
+
+Dsaheader         | 267  | bin          | OpenPGP DSA signature of the header (if thus signed)
+Longsigsize       | 270  | int64        | Header+payload size if > 4GB.
```
Comment (DemiMarie) [inline docs/manual/tags.md:274]: Is this the size of the compressed or uncompressed payload?

File: docs/manual/tags.md [hunk 4]
```diff
@@ -344,259 +182,189 @@ denotes the scriptlet interpreter and possible arguments, and flags tag
...
+Payloaddigest     | 5092 | string array | Cryptographic digest of the compressed payload.
+Payloaddigestalgo | 5093 | int32        | ID of the payload digest algorithm.
+Payloaddigestalt  | 5097 | string array | Cryptographic digest of the uncompressed payload.
+Rsaheader         | 268  | bin          | OpenPGP RSA signature of the header (if thus signed).
+Sha1header        | 269  | string       | SHA1 digest of the header.
+Sha256header      | 273  | string       | SHA256 digest of the header.
+Siggpg            | 262  | bin          | OpenPGP DSA signature of the header+payload (if thus signed).
```
Comment (DemiMarie) [inline docs/manual/tags.md:281]:
```suggestion
Siggpg            | 262  | bin          | OpenPGP DSA or EdDSA signature of the header+payload (if thus signed).
```

File: docs/manual/tags.md [hunk 5]
```diff
@@ -344,259 +182,189 @@ denotes the scriptlet interpreter and possible arguments, and flags tag
...
+Transfiletriggerscriptflags | 5078 | int32 array
+Transfiletriggerscriptprog  | 5077 | string array
+Transfiletriggerscripts     | 5076 | string array
+Transfiletriggerversion     | 5081 | string array
 
-* Sigsize (257) int32
+## Signatures and digests
 
-  Header+payload size.
+[Signatures](signatures.md) allow to verify the origin of a package.
```
Comment (DemiMarie) [inline docs/manual/tags.md:271]: Not all of these tags are actually signatures.  There should be a separation between cryptographic signatures, and entries that just happen to be in the signature header.

File: docs/manual/tags.md [hunk 6]
```diff
@@ -344,259 +182,189 @@ denotes the scriptlet interpreter and possible arguments, and flags tag
...
+Payloaddigest     | 5092 | string array | Cryptographic digest of the compressed payload.
+Payloaddigestalgo | 5093 | int32        | ID of the payload digest algorithm.
+Payloaddigestalt  | 5097 | string array | Cryptographic digest of the uncompressed payload.
+Rsaheader         | 268  | bin          | OpenPGP RSA signature of the header (if thus signed).
+Sha1header        | 269  | string       | SHA1 digest of the header.
+Sha256header      | 273  | string       | SHA256 digest of the header.
+Siggpg            | 262  | bin          | OpenPGP DSA signature of the header+payload (if thus signed).
+Sigmd5            | 261  | bin          | MD5 digest of the header+payload.
+Sigpgp            | 259  | bin          | OpenPGP RSA signature of the header+payload (if thus signed).
+Sigsize           | 257  | int32        | Header+payload size.
```
Comment (DemiMarie) [inline docs/manual/tags.md:284]: Compressed or uncompressed size?  Also, there are tags for just the size of the payload.

File: docs/manual/tags.md [hunk 7]
```diff
@@ -344,259 +182,189 @@ denotes the scriptlet interpreter and possible arguments, and flags tag
...
+Transfiletriggerscripts     | 5076 | string array
+Transfiletriggerversion     | 5081 | string array
 
-* Sigsize (257) int32
+## Signatures and digests
 
-  Header+payload size.
+[Signatures](signatures.md) allow to verify the origin of a package.
+
+Dsaheader         | 267  | bin          | OpenPGP DSA signature of the header (if thus signed)
```
Comment (DemiMarie) [inline docs/manual/tags.md:273]:
```suggestion
Dsaheader         | 267  | bin          | OpenPGP DSA or EdDSA signature of the header (if thus signed)
```

File: docs/manual/tags.md [hunk 8]
```diff
@@ -344,259 +182,189 @@ denotes the scriptlet interpreter and possible arguments, and flags tag
...
+Oldenhancesversion | 1160 | string array
+Oldenhancesflags   | 1161 | int32 array
+Oldfilenames       | 1027 | string array
+Oldsuggestsname    | 1156 | string array
+Oldsuggestsversion | 1157 | string array
+Oldsuggestsflags   | 1158 | int32 array
+Patchesflags       | 1134 | int32 array
+Patchesname        | 1133 | string array
+Patchesversion     | 1135 | string array
+Pubkeys            | 266  | string array
```
Comment (DemiMarie) [inline docs/manual/tags.md:339]: This tag is actually used by RPM internally.  It is ignored if found in the signature header of a package, but if it is found in the main header of a package, it adds a key to the RPM database.

Comment (pmatilai) [reply docs/manual/tags.md:339]: That it is actually used by rpm is true (my bad in the original doc), but the rest is not for any remotely relevant version (rpm 4.4.x behaved somewhat like that). When you don't know for sure, at the very least add an "I think" disclaimer or such, misinformation is bad for everybody.

It's not looked for or used from package files at all, and certainly doesn't "add a key" to the rpmdb. Rpm only ever looks for RPMTAG_PUBKEYS in gpg-pubkey "packages" in the rpmdb.

Comment (DemiMarie) [reply docs/manual/tags.md:339]: Sorry, I should have said, “when found in a gpg-pubkey package”.

File: docs/manual/tags.md (hunk 2 above)
Comment (ffesti) [reply docs/manual/tags.md]: Redone.

File: docs/manual/tags.md [hunk 9]
```diff
@@ -37,6 +37,8 @@ While not technically a type on its own there are several mappings. These consis
 
 ## Base package tags
 
+Tag Name    | Value| Type       | Description
+------------|------|------------|------------
```
[the remark this answers is not in the document]
Comment (ffesti) [reply docs/manual/tags.md:39]: Yeah, added a lot more.

File: docs/manual/tags.md (hunk 8 above)
Comment (ffesti) [reply docs/manual/tags.md]: Moved to the Internal / Special section.

File: docs/manual/tags.md [hunk 10]
```diff
@@ -2,7 +2,38 @@
...
+
+`int8` and `int16` are rarely used. Most integer tags are `int32` with a few `int64` replacements where 32 bits were just not enough. These being unsigned limits time stamps to 1970-01-01 and later.
+
+Additionally there is a `char` datatype that is also only used once.
+
+There is a `bin` datatype for arbitrary data that is basically an `char` array.
+
+There are two string types: Plain `string` with is zero terminated and of arbitary length (within the header size restriction).
+
+`i18nstring`s are translated to the requested local when queried.
```
[the remark this answers is not in the document]
Comment (ffesti) [reply docs/manual/tags.md]: Well, there is a section on arrays in general just below. I don't think I need to treat string arrays special here.

File: docs/manual/tags.md [hunk 11]
```diff
@@ -182,30 +182,44 @@ denotes the scriptlet interpreter and possible arguments, and flags tag
...
 Pretransflags | 5024 | int32
 Pretransprog  | 1153 | string array
 
+`%preun` is run before the package gets removed
+
 Preun      | 1025 | string
 Preunflags | 5022 | int32
 Preunprog  | 1087 | string array
 
+`%verify` script is executed when the package is verified (e.g. with `rpm -V`)
```
[the remark this answers is not in the document]
Comment (ffesti) [reply docs/manual/tags.md:233]: Ok, re-worked that section. I "rightbefore" and "right after" as it indicated as those scripts are run directly as part of the package being installed or removed while the transaction scriptlets are executed at a different time.

**Pull request merged**